src/serializationStrategies/XpathRangeSelector/index.ts

```diff
@@ -33,6 +35,11 @@ export function isLoadable(highlighter: Highlighter, data: IData): boolean {
     return false;
   }
 
+  // Remove all highlights that are not matching current highlighter before getting the range for the data
+  removeAllHighlights(
+    referenceElement,
+    (element: HTMLElement) => !highlighter.getHighlight(element.getAttribute(DATA_ID_ATTR) || ''));
```

Comment:
> isn't this removing actual highlights from the document?

it is and it is probably not the best idea considering that this will run for every highlight

> why are you doing this?

In this case:
1. you have a highlight in a page
2. you search for a phrase that contains this highlight
3. search highlights is correctly mounted and displayed over the normal highlight
4. you refresh a page
5. search highlights is mounted first and then highlights fails to load

Adding this fixes point 5

Maybe this is failing because these util functions: `isHighlight`, `isTextHighlight`, `isTextOrTextHighlight`, `isElementNotHighlight` does not care if the element is a highlight from the specific `highlighter` and returns `null` in cases when they match a search highlight?

Debugging this is not trivial for me so if you have any ideas please share

Reply:
the xpath functions are written in such a way that they ignore highlight elements, so it should not matter if there are already highlights in the document, the mounting should still work. its possible that you've found a situation in which the logic is broken. i'd approach that by trying to recreate your situation in a new unit test for the xpath helpers

Reply:
I've added a test that is working now but is failing when this fragment is deleted.

Malar will create a separate card for this issue and I think that it isn't a priority for now so I'll let it sit for a while